sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2019 Tecnativa - Ernesto Tejeda
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+odoo.define('sale_layout_category_hide_detail.boolean_fa_icon_widget', function (require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var registry = require('web.field_registry');
+
+    var BooleanFaIconWidget = AbstractField.extend({
+        className: 'o_boolean_fa_icon_widget',
+        events: {
+            'click': '_toggleValue'
+        },
+        supportedFieldTypes: ['boolean'],
+
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+
+        /**
+         * A boolean field is always set since false is a valid value.
+         *
+         * @override
+         */
+        isSet: function () {
+            return true;
+        },
+
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * Render font-awesome icon based on state
+         *
+         * @override
+         * @private
+         */
+        _render: function () {
+            var options = this.attrs.options;
+            var fa_class = this.value ? options.fa_icons.icon_true : options.fa_icons.icon_false;
+            var tip = this.value ? options.terminology.hover_true : options.terminology.hover_false;
```

Comment:
Again, if we really are going to keep this widget, these things need default values.